state-chain/pallets/cf-broadcast/src/lib.rs

```diff
@@ -501,20 +481,26 @@ pub mod pallet {
 					(broadcast_id, initiated_at),
 				);
 
-				let broadcast_attempt = BroadcastAttempt::<T, I> {
+				// Store the new broadcast data so it is ready to be broadcasted.
+				let broadcast_data = BroadcastData::<T, I> {
 					broadcast_id,
 					transaction_payload: T::TransactionBuilder::build_transaction(&signed_api_call),
 					threshold_signature_payload,
 					transaction_out_id,
 				};
+				// This write is required in-case if no nominee can be selected in the first try.
```

Comment:
See my other comment - maybe we don't need this write if we flatten the nominee into the `BroadcastData` struct.